File: java/parser/src/main/java/gov/lbl/parser/parser/Logical_expressionVisitor.java
package gov.lbl.parser.parser;

import java.util.List;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Logical_expression;
import gov.lbl.parser.domain.Logical_term;

import static java.util.stream.Collectors.toList;

public class Logical_expressionVisitor extends modelicaBaseVisitor<Logical_expression> {
    @Override
    public Logical_expression visitLogical_expression(modelicaParser.Logical_expressionContext ctx) {

        Logical_termVisitor logical_termVisitor = new Logical_termVisitor();
        List<Logical_term> logical_term_list = ctx.logical_term()
                .stream()
                .map(l_term -> l_term.accept(logical_termVisitor))
                .collect(toList());
        
        return new Logical_expression(logical_term_list);
    }
}

File: jsParser/parser/Logical_expressionVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Logical_expression = require('../domain/Logical_expression');

const Logical_termVisitor = require('./Logical_termVisitor');

class Logical_expressionVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitLogical_expression(ctx) {
        var logical_term_list = [];

        if (ctx.logical_term()) {
            var logical_termVisitor = new Logical_termVisitor.Logical_termVisitor();
            ctx.logical_term().forEach(term => {
                logical_term_list.push(logical_termVisitor.visitLogical_term(term));
            });
        }
        
        return new Logical_expression.Logical_expression(logical_term_list);
    }
};

Logical_expressionVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitLogical_expression = this.visitLogical_expression;
exports.Logical_expressionVisitor = Logical_expressionVisitor;